### DISCUSSION TITLE
Linux Docker issue - "Couldn't connect to Docker daemon at http+docker://localhost"

### DISCUSSION THREAD

**Discussion by lpatmo** [codebuddies/backend#174] [Q&A] [answered]
@peoray is running into this after installing Docker on Linux:

Step 1: https://docs.docker.com/engine/install/ubuntu/
Step 2: https://docs.docker.com/compose/install/
![image](https://user-images.githubusercontent.com/4512699/92313821-dc5c9d00-ef84-11ea-9637-1835c37e5fce.png)

The error:

![image](https://user-images.githubusercontent.com/4512699/92313808-b8995700-ef84-11ea-951b-d8d4306c93bf.png)

**Comment by lpatmo** [comment]
I am reading through https://github.com/docker/compose/issues/4181 and... maybe `service docker restart` can help?

**Comment by lpatmo** [comment]
@peoray is now getting this error:
![image](https://user-images.githubusercontent.com/4512699/92313936-47f33a00-ef86-11ea-8e8f-f05efe61bdde.png)

**Reply by lpatmo** [reply]
This is after @peoray ran `export DOCKER_HOST=127.0.0.1`

**Comment by BethanyG** [comment, marked as answer]
Ok - so it _smells_ like a permissions problem.  Below are several SO posts, et. al.  While they talk about `docker.sock`, methinks this error might be related:

-  [Medium:  Couldn't Connect Docker](https://medium.com/@ibrahimgunduz34/if-you-faced-an-issue-like-couldnt-connect-to-docker-daemon-at-http-docker-localunixsocket-is-27b35f17d09d)
-  [SO:  Couldn't Connect to docker from docker-compose](https://stackoverflow.com/questions/29101043/cant-connect-to-docker-from-docker-compose)
-  [SO:  Couldn't Connect Docker](https://stackoverflow.com/questions/50979424/couldnt-connect-to-docker-daemon-at-httpdocker-localhost-with-docker-compose)
-  [GitHub Couldn't Connect Docker](https://github.com/docker/compose/issues/4181)

All of them talk about the perils of running docker-compose with `sudo`.  They also talk about making sure docker has permissions to everywhere it should...and to have env vars set properly.  Since I am not on linux, I can't test each specifically, but I'd do a general rundown of permissions and env vars just to make sure they are all as expected.

**Reply by peoray** [reply]
Thank you Bethany. Your guess was right, it was a permission issue. Had to `sudo` to the command to get it working :)

**Reply by BethanyG** [reply]
So happy you got the problem solved!  YAY!
